openmrs-keycloak-smart-auth/src/main/java/org/openmrs/contrib/keycloak/smart/auth/provider/ExternalAppAuthenticator.java

```diff
@@ -62,6 +70,36 @@ public void authenticate(AuthenticationFlowContext context) {
             applicationId = DEFAULT_APPLICATION_ID;
         }
 
+        String scope = context.getAuthenticationSession().getClientNote(OIDCLoginProtocol.SCOPE_PARAM);
+        Map<String, ClientScopeModel> defaultScopes = context.getAuthenticationSession().getClient()
+                .getClientScopes(true, true);
+
+        if (scope == null) {
+            scope = "";
+        }
+
+        List<String> scopes = Arrays.stream(scope.split("\\s+")).collect(Collectors.toList());
+        String supportedParams = context.getAuthenticatorConfig().getConfig()
+                .get(SmartLaunchAuthenticatorFactory.CONFIG_EXTERNAL_SMART_LAUNCH_SUPPORTED_PARAMS);
+        if (supportedParams == null || supportedParams.isEmpty()) {
+            context.attempted();
+            return;
+        }
+
+        List<String> params = Arrays.stream(supportedParams.split("\\s+")).collect(Collectors.toList());
+        boolean foundMatch = false;
+        for (String param : params) {
+            if (scopes.contains(LAUNCH_SCOPE_PREFIX + param) ||
+                    defaultScopes.containsKey(LAUNCH_SCOPE_PREFIX + param)) {
+                foundMatch = true;
+            }
+        }
+
+        if (!foundMatch) {
+            context.attempted();
+            return;
+        }
+
```

Comment:
@ibacher So this code doesn't seems to work. Can you point out what needs to be changed?

Reply:
When you tried this out, did you add any scopes to the `smart-launch-supported-params` configuration? The default value, `null`, means that this will exit on line 86.

Reply:
Yes so I left it empty (null) but then also I was redirected to the patient selector page.

Reply:
Oh! I just realised what the error is. For some reason these `ExternalAuthenticator` classes got left in-place but, in fact, we're not using them. So these changes need to be added to the `SmartLaunchAuthenticator` and `SmartLaunchAuthenticatorFactory`. I just pushed a commit that removes the other classes.